src/components/clusterDetail/ConsoleModal.tsx

```diff
@@ -77,34 +81,102 @@ const getHostIPs = (cluster: Cluster): { [key: string]: string } => {
   return hostIPAddresses;
 };
 
+const getVirtualIPs = ({
+  apiVip,
+  ingressVip,
+  highAvailabilityMode,
+  machineNetworkCidr = '',
+  hosts = [],
+  hostNetworks = [],
+}: Cluster) => {
+  const vipValues = { apiVip, ingressVip };
+  if (isSingleNodeCluster({ highAvailabilityMode }) && machineNetworkCidr?.length > 1) {
+    // We assume at this point the server will
+    // populate this value with either an IPv4 or IPv6 address
+    const isIPv4 = isCIDR.v4(machineNetworkCidr);
+    const network = isIPv4 ? new Address4(machineNetworkCidr) : new Address6(machineNetworkCidr);
+    const hostIds =
+      hostNetworks?.find((hostNetwork) => hostNetwork.cidr === machineNetworkCidr)?.hostIds || [];
+    // According to BE team, if it's a SNO cluster, there should be only one host in the cluster
+    // network with status 'installed'
+    const host = hostIds
+      ?.map((hostId) => findHostById(hosts, hostId))
```

Comment:
this can be done via `Array.filter()`

Reply:
it is exactly what it is doing. think about this function as a common operation you would like to perform on the hosts list